### ISSUE TITLE
[Suggestion] Settings menu for enabling / disabling features

### ISSUE THREAD

**Issue by FlamedDogo99** [FlamedDogo99/EaglerMobile#25]
**Suggestion**
Currently the user has no way to customize their experience with Eagler Mobile. This could be changed by adding a custom menu that allows the user to control parts of the script.
**Additional Information**
The following features would be nice to have:
 - [ ] A toggle for Safari’s file input fix
 - [ ] A toggle for the pointerlock and full screen fake API’s
 - [ ] A toggle for manually setting keyboard mode
      - It could be good to give a warning when switching these
 - [ ] A slider for inGame mobile control opacity
 - [ ] A slider for inMenu mobile control opacity
 - [ ] A menu for editing UI
   - [ ] Setting button size
   - [ ] Setting button placement + right/left handed toggle
   - [ ] Joystick / D-pad toggle
 - [ ] A slider for touch sensitivity
 - [ ] A slider for long hold detection duration

**Comment by FlamedDogo99** [comment]
Alright here’s some thoughts on this, because this issue will likely be my next focus.

This will either be a check in every function for a setting, or programmatically redefining functions when a setting changes. It would be nice to have as little code running as possible on top of the client, however programmatically redefining things will likely be more error prone.

In regards to UI, I’m either going to copy pre-established styles from PE or legacy console edition, or if #28 finishes it’s mobile controls interface it may be good to keep consistency across versions and base it off of that.

**Comment by FlamedDogo99** [comment]
It would also be good to address how we are storing information used by the script. 
The current `clientWindow` system sucks and is generally unsafe if we are using the script on untrusted websites.
The problem is making sure that the userscript has access to the same variables and functions that the injected functions do. 
We could just directly append the script to the DOM, but that would be redundant if the user already is running it from a script tag.
One option would be to detect and delete the containing script, but that doesn’t feel like a good solution
There’s got to be a better way to allow the event listeners and redefined functions to read from the userscript’s internal state, but I currently have no idea what that would be.

**Comment by FlamedDogo99** [comment]
This could could be a solution:
```js
if (typeof window.wrappedJSObject == "object") {
  console.log("wrappedJSObject detected");
  location.href = "javascript:(" + encodeURI(arguments.callee.toSource()) + ")();";
  return;
}
```
It throws an error on mobile, so it might need a `try{ }catch(e){ }`

**Comment by FlamedDogo99** [comment]
Closed in favor of #34
